app/controllers/merge_requests_controller.rb

```diff
@@ -132,7 +132,11 @@ def validate_response
 
     if page == "confirm_telephone_number"
       if merge_request_params[:telephone_number_correct].blank? && merge_request_params[:new_telephone_number].blank?
-        @merge_request.errors.add(:telephone_number_correct, I18n.t("validations.merge_request.telephone_number_correct_blank"))
+        if @merge_request.absorbing_organisation.phone.present?
+          @merge_request.errors.add(:telephone_number_correct, I18n.t("validations.merge_request.telephone_number_correct_blank"))
+        else
+          @merge_request.errors.add(:telephone_number_correct, I18n.t("validations.merge_request.new_telephone_number_blank"))
+        end
         render previous_template
```

Comment:
`render previous_template, status: :unprocessable_entity` ? same for the one below?